server-adapters/resteasy-reactor-netty/src/test/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStreamTest.java

```diff
@@ -0,0 +1,306 @@
+import reactor.netty.channel.AbortedException;
+import reactor.netty.http.server.HttpServerResponse;
+import reactor.netty.http.server.WebsocketServerSpec;
+import reactor.netty.http.websocket.WebsocketInbound;
+import reactor.netty.http.websocket.WebsocketOutbound;
+import reactor.test.StepVerifier;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import static org.jboss.resteasy.plugins.server.reactor.netty.ChunkOutputStream.RESPONSE_WRITE_ABORTED_ON_CANCEL;
+import static org.jboss.resteasy.plugins.server.reactor.netty.ChunkOutputStream.RESPONSE_WRITE_ABORTED_ON_DISCARD;
+
+public class ChunkOutputStreamTest {
+
+    @Test
+    public void testAsyncWrite_withErrorOnSend() {
+        final Sinks.Empty<Void> completionSink = Sinks.empty();
+
+        final TestHttpServerResponse testHttpServerResponse = new ErrorHttpServerResponse();
+        final ReactorNettyHttpResponse httpServerResponse = new ReactorNettyHttpResponse(
+                HttpMethod.POST,
+                testHttpServerResponse,
+                completionSink
+        );
+        final ChunkOutputStream chunkOutputStream = new ChunkOutputStream(
+                httpServerResponse,
+                testHttpServerResponse,
+                completionSink
+        );
+
+        final byte[] errorBytes = "ERROR".getBytes(StandardCharsets.UTF_8);
+        final CompletableFuture<Void> asyncWriteFuture = chunkOutputStream.asyncWrite(
+                errorBytes,
+                0,
+                errorBytes.length
+        );
+
+        StepVerifier.create(Mono.fromCompletionStage(asyncWriteFuture).then(completionSink.asMono()))
+                .verifyError(AbortedException.class);
+    }
+
+    @Test
+    public void testAsyncWrite_forErrorOnDiscard() {
+        final Sinks.Empty<Void> completionSink = Sinks.empty();
+
+        final TestHttpServerResponse testHttpServerResponse = new DiscardingHttpServerResponse();
+        final ReactorNettyHttpResponse httpServerResponse = new ReactorNettyHttpResponse(
+                HttpMethod.POST,
+                testHttpServerResponse,
+                completionSink
+        );
+        final ChunkOutputStream chunkOutputStream = new ChunkOutputStream(
+                httpServerResponse,
+                testHttpServerResponse,
+                completionSink
+        );
+
+        final byte[] errorBytes = "DISCARD".getBytes(StandardCharsets.UTF_8);
+        final CompletableFuture<Void> asyncWriteFuture = chunkOutputStream.asyncWrite(
+                errorBytes,
+                0,
+                errorBytes.length
+        );
+
+        StepVerifier.create(Mono.fromCompletionStage(asyncWriteFuture).then(completionSink.asMono()))
```

Comment:
This looks nice.  My only concern is that this test has the possibility of not timing out.  There are multiple ways to deal with this.  On option is to put a timeout on the `Mono` you are passing the `StepVerifier#create`.